PWGHF/HFC/TableProducer/HFCorrelatorD0Hadrons.cxx

```diff
@@ -0,0 +1,491 @@
+      if (cutYCandMax >= 0. && std::abs(YD0(candidate1)) > cutYCandMax) {
+        continue;
+      }
+      if (cutPtCandMin >= 0. && candidate1.pt() < cutPtCandMin) {
+        continue;
+      }
+      // check decay channel flag for candidate1
+      if (!(candidate1.hfflag() & 1 << DecayType::D0ToPiK)) {
+        continue;
+      }
+
+      // ========================== Define parameters for soft pion removal ================================
+      double massPi = RecoDecay::getMassPDG(kPiPlus);
+      double massK = RecoDecay::getMassPDG(kKPlus);
+      double EPiProng0 = RecoDecay::e(candidate1.pVectorProng0(), massPi);
+      double EPiProng1 = RecoDecay::e(candidate1.pVectorProng1(), massPi);
+      double EKProng0 = RecoDecay::e(candidate1.pVectorProng0(), massK);
+      double EKProng1 = RecoDecay::e(candidate1.pVectorProng1(), massK);
+
+      // ========================== trigger efficiency ================================
+      double efficiencyWeight = 1.;
+      if (flagApplyEfficiency) {
+        efficiencyWeight = 1. / efficiencyDmeson->at(o2::analysis::findBin(bins, candidate1.pt()));
+      }
+      // ========================== Fill mass histo  ================================
+      if (candidate1.isSelD0() >= selectionFlagD0) {
+        registry.fill(HIST("hMass"), InvMassD0(candidate1), candidate1.pt(), efficiencyWeight);
+        registry.fill(HIST("hMass1D"), InvMassD0(candidate1), efficiencyWeight);
+        registry.fill(HIST("hMassD01D"), InvMassD0(candidate1), efficiencyWeight);
+      }
+      if (candidate1.isSelD0bar() >= selectionFlagD0bar) {
+        registry.fill(HIST("hMass"), InvMassD0bar(candidate1), candidate1.pt(), efficiencyWeight);
+        registry.fill(HIST("hMass1D"), InvMassD0bar(candidate1), efficiencyWeight);
+        registry.fill(HIST("hMassD0bar1D"), InvMassD0bar(candidate1), efficiencyWeight);
+      }
+      // ========================== Fill general histos ================================
+      registry.fill(HIST("hPtCand"), candidate1.pt());
+      registry.fill(HIST("hPtProng0"), candidate1.ptProng0());
+      registry.fill(HIST("hPtProng1"), candidate1.ptProng1());
+      registry.fill(HIST("hEta"), candidate1.eta());
+      registry.fill(HIST("hPhi"), candidate1.phi());
+      registry.fill(HIST("hY"), YD0(candidate1));
+      registry.fill(HIST("hSelectionStatus"), candidate1.isSelD0bar() + (candidate1.isSelD0() * 2));
+
+      // ================================================================================= D-h correlation dedicated section =====================================================
+
+      // ========================== track loop starts here ================================
+      for (const auto& track : tracks) {
+        registry.fill(HIST("hTrackCounterData"), 1); // fill total no. of tracks
+        // Remove D0 daughters by checking track indices
+        if ((candidate1.index0Id() == track.mRowIndex) || (candidate1.index1Id() == track.mRowIndex)) {
```

Comment:
```suggestion
        if ((candidate1.index0Id() == track.globalIndex()) || (candidate1.index1Id() == track.globalIndex())) {
```
I think here you should use globalIndex instead of mRowIndex, since the latter starts from 0 at each collision, and won't correspond to candidate1.index0Id() or candidate1.index1Id(), which are instead incremental.
At least, I'm doing and this seems to be the behaviour. This should be fixed also on the D+-hadron code (I'll inform Shyam)